imap_processing/idex/idex_l1a.py

```diff
@@ -131,9 +131,9 @@ def __init__(self, packet_file: Union[str, Path], data_version: str) -> None:
 
         dust_events = {}
         for packet in decom_packet_list:
-            if "IDX__SCI0TYPE" in packet.data:
-                scitype = packet.data["IDX__SCI0TYPE"].raw_value
-                event_number = packet.data["IDX__SCI0EVTNUM"].derived_value
+            if "IDX__SCI0TYPE" in packet.user_data:
```

Comment:
We should look up in the packet in general, not just the user_data. The user_data adds extra overhead to the lookup because it creates a new dictionary under the hood.
```suggestion
            if "IDX__SCI0TYPE" in packet:
```